src/location.rs

```diff
@@ -1,40 +1,129 @@
+use crate::Span;
+use std::cell::RefCell;
+use std::rc::Rc;
+
 /// Represents the source location (row, column) of this element in the original Ion text.
 ///
 /// The source location metadata is primarily intended for error reporting and debugging purposes,
 /// helping applications provide meaningful feedback to users about the source of issues.
-///
-/// # Returns
-/// * `Some((row, column))` - Position where this element was found in the source text
-/// * `None` - Location information is not available
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Default)]
 pub struct SourceLocation {
-    location: Option<(usize, usize)>,
+    location: (usize, usize),
```

Comment:
🪧 Since `(0, 0)` is not a valid row/column, we don't need to have the `Option`. Removing the option reduces the size of this type from 24 bytes to 16 bytes (on 64-bit systems).